File: dice/golang/sol/apigear/game/dice_test.go
package game

import "testing"

type randomMock struct {
	count int
}

func (d *randomMock) Generate(min int, max int) int {
	d.count++
	return min
}

func TestDiceToss(t *testing.T) {
	r := randomMock{0}
	d := NewDice(&r)
	if r.count != 0 {
		t.Fail()
	}
	d.Toss()
	if r.count != 1 {
		t.Fail()
	}
}

File: dice/typescript/sol/src/apigear_game/dice.test.ts
import { IRandom } from "./api";
import { Dice } from "./dice";

class MockRandom implements IRandom {
  count = 0;
  generate(min: number, max: number): number {
    this.count++;
    return min;
  }
}

test("Dice.toss", () => {
  const rand = new MockRandom();
  const dice = new Dice(rand);
  expect(rand.count).toBe(0);
  dice.toss();
  expect(rand.count).toBe(1);
});
